app/src/main/java/org/jellyfin/androidtv/ui/home/HomeFragmentLiveTVRow.kt

```diff
@@ -21,12 +15,11 @@ import org.jellyfin.androidtv.ui.presentation.CardPresenter
 import org.jellyfin.androidtv.ui.presentation.GridButtonPresenter
 import org.jellyfin.apiclient.model.dto.BaseItemDto
 import org.jellyfin.apiclient.serialization.GsonJsonSerializer
-import org.koin.core.KoinComponent
-import org.koin.core.inject
-
-class HomeFragmentLiveTVRow(val activity: Activity) : HomeFragmentRow(), OnItemViewClickedListener, KoinComponent {
-	private val serializer: GsonJsonSerializer by inject()
 
+class HomeFragmentLiveTVRow(
+	private val activity: Activity,
+	private val serializer: GsonJsonSerializer
```

Comment:
The sample I saw seemed to suggest that we should be using `by inject()` here as the default value and would only need to override for special conditions like tests. Although their docs do not seem to reflect the suggestions in their post so :man_shrugging:

Reply:
we can only use the `inject` and `get` functions when the current context is Koin aware. This can be done by extending the KoinComponent class or is automatically done for Activities, Fragments and Services with the Android module.
For other places passing dependencies via constructor is recommended. The apiclient uses that approach in the openapi-generator.

> use constructor style injection as much as possible.

- Blog post above